### PR TITLE
Allow partners to configure ISV tracking GUID

### PR DESCRIPTION
**Pull request by gossion**
- [x] Please check this box and fill the data as below once you have run the unit tests.
      Code coverage before your change: `865 examples, 0 failures. 3440 / 3493 LOC (98.48%) covered.`
      Code coverage with your change:   `868 examples, 0 failures. 3444 / 3497 LOC (98.48%) covered.`

  ```
  pushd src/bosh_azure_cpi
    ./bin/test-unit
  popd
  ```

  NOTE: Please see how to setup dev environment and run unit tests in docs/development.md.

### Changelog

* Allow partners to configure ISV tracking GUID

### CODE REVIEW

File: src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb [hunk 1]
```diff
@@ -2417,7 +2417,10 @@ def http_post(url, body = nil, params = {}, retry_after = 5)
     end
 
     def merge_http_common_headers(request)
-      request['User-Agent'] = USER_AGENT_FOR_REST
+      user_agents = ["#{USER_AGENT_FOR_REST}/#{Bosh::AzureCloud::VERSION}"]
+      isv_tracking_guid = @azure_properties['isv_tracking_guid'].nil? ? DEFAULT_ISV_TRACKING_GUID : @azure_properties['isv_tracking_guid']
```
Comment (bingosummer) [inline src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb]: How about `isv_tracking_guid = @azure_properties.fetch('isv_tracking_guid', DEFAULT_ISV_TRACKING_GUID)`?

File: jobs/azure_cpi/templates/cpi.json.erb [hunk 2]
```diff
@@ -75,6 +75,10 @@
     params['cloud']['properties']['azure']['azure_stack']['endpoint_prefix'] = p('azure.azure_stack.endpoint_prefix')
   end
 
+  if_p('azure.isv_tracking_guid') do |isv_tracking_guid|
+    params['cloud']['properties']['azure']['isv_tracking_guid'] = isv_tracking_guid
```
Comment (bingosummer) [inline jobs/azure_cpi/templates/cpi.json.erb]: How about checking whether it's really a GUID? At least the length should be 36?

File: src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb [hunk 3]
```diff
@@ -2417,7 +2417,10 @@ def http_post(url, body = nil, params = {}, retry_after = 5)
...
+      user_agents.push("pid-#{isv_tracking_guid}")
```
Comment (bingosummer) [inline src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb]: The format has to be "pid-#{isv_tracking_guid}"? Or defined by CPI?

File: src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb (hunk 1 above)
Comment (gossion) [reply src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb]: good suggestion.

File: src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb (hunk 3 above)
Comment (gossion) [reply src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb]: has to be 'pid-xxx'

Comment (bingosummer) [reply src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb]: Then it's better to add a comment to link to the doc.

File: jobs/azure_cpi/templates/cpi.json.erb (hunk 2 above)
Comment (gossion) [reply jobs/azure_cpi/templates/cpi.json.erb]: thanks. will change.

File: src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb (hunk 3 above)
Comment (gossion) [reply src/bosh_azure_cpi/lib/cloud/azure/azure_client2.rb]: The link is only for internal currently. Can't add the link.

**Pull request merged**